pkg/apis/pipeline/v1alpha1/cluster_resource_test.go

```diff
@@ -0,0 +1,183 @@
+			Type:   PipelineResourceTypeCluster,
+			URL:    "http://10.10.10.10",
+			CAData: []byte("my-cluster-cert"),
+			Token:  "my-token",
+		},
+	}, {
+		desc: "resource with password instead of token",
+		resource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster-resource",
+				Namespace: "foo",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeCluster,
+				Params: []Param{{
+					Name:  "url",
+					Value: "http://10.10.10.10",
+				}, {
+					Name:  "cadata",
+					Value: "bXktY2x1c3Rlci1jZXJ0Cg",
+				}, {
+					Name:  "username",
+					Value: "user",
+				}, {
+					Name:  "password",
+					Value: "pass",
+				},
+				},
+			},
+		},
+		want: &ClusterResource{
+			Name:     "test-cluster-resource",
+			Type:     PipelineResourceTypeCluster,
+			URL:      "http://10.10.10.10",
+			CAData:   []byte("my-cluster-cert"),
+			Username: "user",
+			Password: "pass",
+		},
+	}, {
+		desc: "set insecure flag to true when there is no cert",
+		resource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster-resource",
+				Namespace: "foo",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeCluster,
+				Params: []Param{{
+					Name:  "url",
+					Value: "http://10.10.10.10",
+				}, {
+					Name:  "token",
+					Value: "my-token",
+				},
+				},
+			},
+		},
+		want: &ClusterResource{
+			Name:     "test-cluster-resource",
+			Type:     PipelineResourceTypeCluster,
+			URL:      "http://10.10.10.10",
+			Token:    "my-token",
+			Insecure: true,
+		},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			got, err := NewClusterResource(c.resource)
+			if err != nil {
+				t.Errorf("Test: %q; TestNewClusterResource() error = %v", c.desc, err)
+			}
+			if d := cmp.Diff(got, c.want); d != "" {
+				t.Errorf("Diff:\n%s", d)
+			}
+		})
+	}
+}
+
+func TestSecrets(t *testing.T) {
+	for _, c := range []struct {
+		desc     string
+		resource *PipelineResource
+		want     *ClusterResource
+	}{{
+		desc: "basic resource with secrets",
+		resource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster-resource",
+				Namespace: "foo",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeCluster,
+				Params: []Param{{
+					Name:  "url",
+					Value: "http://10.10.10.10",
+				}},
+				SecretParams: []SecretParam{{
+					FieldName:  "cadata",
+					SecretKey:  "cadatakey",
+					SecretName: "secret1",
+				}, {
+					FieldName:  "token",
+					SecretKey:  "tokenkey",
+					SecretName: "secret1",
+				}},
+			},
+		},
+		want: &ClusterResource{
+			Name: "test-cluster-resource",
+			Type: PipelineResourceTypeCluster,
+			URL:  "http://10.10.10.10",
+			Secrets: []SecretParam{{
+				FieldName:  "cadata",
+				SecretKey:  "cadatakey",
+				SecretName: "secret1",
+			}, {
+				FieldName:  "token",
+				SecretKey:  "tokenkey",
+				SecretName: "secret1",
+			}},
+		},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
```

Comment:
if there's only one case in this test, i feel like it's clearer not to make it table driven

Reply:
will add the secret test case in the other test